yaptide/routes.py

```diff
@@ -1,22 +1,40 @@
-from flask import request, json, jsonify
+from flask import request, json, make_response
 from flask_api import status as api_status
-from flask_restful import Resource, reqparse, fields, marshal_with, abort
-from warnings import resetwarnings
+from flask_restful import Resource
 
-from werkzeug.datastructures import MultiDict
 from yaptide.persistence.database import db
-from yaptide.persistence.models import ExampleUserModel
+from yaptide.persistence.models import UserModel
 
 from yaptide.simulation_runner.shieldhit_runner import run_shieldhit, celery_app
-from marshmallow import Schema
-from marshmallow import fields as fld
-
 from celery.result import AsyncResult
 
+from marshmallow import Schema, ValidationError
+from marshmallow import fields as fld
+
 from typing import Union, Literal
+from werkzeug.datastructures import MultiDict
+from werkzeug.exceptions import Unauthorized
+from werkzeug import exceptions
+
+from functools import wraps
+
+import datetime
 
 resources = []
 
+
+def requires_auth(f):
+    """Determines if the access token is valid"""
+    @wraps(f)
+    def decorated(*args, **kwargs):
```

Comment:
add some comment explaining the logic of this decorator

Reply:
Decorator was reworked